js/pokemon.js

```diff
@@ -100,44 +101,56 @@ $(document).ready(function() {
  * Set the range of numbers to generate from depending on the Pokemon Generation selected
  */
  
-function setGen(selectedGen) {
-
-    if (selectedGen == 1) {
-        minPokemonNumber = 1;
-        maxPokemonNumber = 151;
-    } else if (selectedGen == 2) {
-        minPokemonNumber = 152;
-        maxPokemonNumber = 251;
-    } else if (selectedGen == 3) {
-        minPokemonNumber = 252;
-        maxPokemonNumber = 386;
-    } else if (selectedGen == 4) {
-        minPokemonNumber = 387;
-        maxPokemonNumber = 493;
-    } else if (selectedGen == 5) {
-        minPokemonNumber = 494;
-        maxPokemonNumber = 649;
-    } else {
-        minPokemonNumber = 1;
-        maxPokemonNumber = 649;
-    }
+function setGen(genToAffect) {
 
-    if (newGen == -1) {
+    genToAffect = Number(genToAffect);
+
+    if (newGen[0] === -1) {
         // first time this has been called, so make the choice active, not just selected
-        document.getElementById('gen' + selectedGen).className += " current";
-        currentGen = selectedGen;
-    } else {
-        document.getElementById('gen' + newGen).className = document.getElementById('gen' + newGen).className.replace('selected','');
+        document.getElementById('gen' + genToAffect).classList.add("current");
+        currentGen = [genToAffect];
+        newGen = [genToAffect];
+        return 0;
+    }
+
+    //Before editing gen selection, we ensure that the user is not about to remove their last gen
+    if (!(newGen.length === 1 && newGen[0] === genToAffect)) {
+
+        //If the user chooses a specific Gen while "All" is active, we unselect "All"
```

Comment:
I think the concept of an "all" generation should be removed, and have all generations instead selected by default. Certainly if the game had been made with multi-generation from the beginning, "All" would never have existed.